src/ui/feed/note/mod.rs

```diff
@@ -914,7 +914,8 @@ fn render_content(
                             .monospace()
                             .italics(),
                         );
-                        if ui.button("Show Post").clicked() {
+                        if ui.button("Show Post").clicked() || app.settings.approve_content_warning
```

Comment:
This code is run every frame rendered. When you turn on your setting, it will keep adding the event.id to the app.approved vector over and over.  [it's ok with clicked() because that only fires for one frame].

You should instead look several lines up and change the 'else if' statement that tests against the app.approved.contains and add `&& !app.settings.approve_content_warning`

Reply:
Thank you very much for precious suggestions and your time in writing your explanations.